### DISCUSSION TITLE
Bail on only the file with the failure

### DISCUSSION THREAD

**Discussion by mattfysh** [vitest-dev/vitest#6594] [Q&A]
Hello, I've implemented my e2e tests using vitest and set `bail: 1`

is there any way to scope the bail to the file where failure occurs? currently when a failure occurs in any of the e2e test files, all e2e tests will finish the current test and then bail

i only want the file with the failure to bail

**Comment by hi-ogawa** [comment]
> is there any way to scope the bail to the file where failure occurs?

`bail` option doesn't work this way, but this might be possible by using `beforeEach` and `afterEach` contexts like this:
https://stackblitz.com/edit/vitest-dev-vitest-zqeahj?file=test%2Fbasic.test.ts

```ts
let bail = false;

beforeEach((ctx) => {
  if (bail) {
    ctx.skip();
  }
})

afterEach((ctx) => {
  if (ctx.task.result?.state === "fail") {
    bail = true;
  }
})
```
